addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -411,3 +411,138 @@ def test_order_to_invoice(self):
         # I confirm the bank statement using Confirm button
 
         self.AccountBankStatement.button_confirm_bank()
+
+    def test_create_from_ui(self):
+        """
+        Simulation of sales coming from the interface, even after closing the session
+        """
+        # I click on create a new session button
+        self.pos_config.open_session_cb()
+
+        current_session = self.pos_config.current_session_id
+        num_starting_orders = len(current_session.order_ids)
+
+        carrot_order = {'data':
+          {'amount_paid': 1.04,
+           'amount_return': 0,
+           'amount_tax': 0.14,
+           'amount_total': 1.04,
+           'creation_date': fields.Datetime.now(),
+           'fiscal_position_id': False,
+           'lines': [[0,
+             0,
+             {'discount': 0,
+              'id': 42,
+              'pack_lot_ids': [],
+              'price_unit': 0.9,
+              'product_id': self.carotte.id,
+              'qty': 1,
+              'tax_ids': [[6, False, self.carotte.taxes_id.ids]]}]],
+           'name': 'Order 00042-003-0014',
+           'partner_id': False,
+           'pos_session_id': current_session.id,
+           'sequence_number': 2,
+           'statement_ids': [[0,
+             0,
+             {'account_id': self.env.user.partner_id.property_account_receivable_id.id,
+              'amount': 1.04,
+              'journal_id': self.pos_config.journal_ids[0].id,
+              'name': fields.Datetime.now(),
+              'statement_id': current_session.statement_ids[0].id}]],
+           'uid': '00042-003-0014',
+           'user_id': self.env.uid},
+          'id': '00042-003-0014',
+          'to_invoice': False}
+
+        zucchini_order = {'data':
+          {'amount_paid': 1.3800000000000001,
+           'amount_return': 0,
+           'amount_tax': 0.18,
+           'amount_total': 1.38,
+           'creation_date': fields.Datetime.now(),
+           'fiscal_position_id': False,
+           'lines': [[0,
+             0,
+             {'discount': 0,
+              'id': 3,
+              'pack_lot_ids': [],
+              'price_unit': 1.2,
+              'product_id': self.courgette.id,
+              'qty': 1,
+              'tax_ids': [[6, False, self.courgette.taxes_id.ids]]}]],
+           'name': 'Order 00043-003-0014',
+           'partner_id': False,
+           'pos_session_id': current_session.id,
+           'sequence_number': self.pos_config.journal_id.id,
+           'statement_ids': [[0,
+             0,
+             {'account_id': self.env.user.partner_id.property_account_receivable_id.id,
+              'amount': 1.3800000000000001,
```

Comment:
And the payment has the same rounding weirdness, which if it really is a bug in ```round_precision()``` makes sense I guess.